src/data/nat/fib.lean

```diff
@@ -47,23 +50,16 @@ def fib (n : ℕ) : ℕ := (fib_aux_stream n).fst
 @[simp] lemma fib_zero : fib 0 = 0 := rfl
 @[simp] lemma fib_one : fib 1 = 1 := rfl
 
-private lemma fib_recurrence_aux {n : ℕ} : (fib_aux_stream (n + 1)).fst = (fib_aux_stream n).snd :=
+private lemma fib_aux_stream_succ {n : ℕ} :
+    fib_aux_stream (n + 1) = fib_aux_step (fib_aux_stream n) :=
 begin
-  change (stream.nth (n + 1) $ stream.iterate _ (0, 1)).fst = _,
-  rw [stream.nth_succ_iterate, stream.map_iterate, stream.nth_map, fib_aux_stream]
+  change (stream.nth _ $ stream.iterate _ _) = fib_aux_step (stream.nth _ $ stream.iterate _ _),
```

Comment:
Perhaps replace some of the `_` back to their values? (The goal isn't to write the shortest term possible; use underscores to hide things that are not helpful.) I think the first arguments of both `stream.nth`s are helpful, and perhaps the `(0,1)` arguments of `stream.iterate`.

Reply:
Sounds good. I went ahead and filled in all the underscores, to make explicit that the two `iterate` calls are the same.